### ISSUE TITLE
Pre-Training BLIP2: RuntimeError: probability tensor contains either `inf`, `nan` or element < 0

### ISSUE THREAD

**Issue by evelinehong** [salesforce/LAVIS#107]
Hey I encounter this issue when I try to pre-train BLIP2 with COCO:
 ```
Traceback (most recent call last):
  File "train.py", line 103, in <module>
    main()
  File "train.py", line 99, in main
    runner.train()
  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/runners/runner_base.py", line 352, in train
    train_stats = self.train_epoch(cur_epoch)
  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/runners/runner_base.py", line 411, in train_epoch
    return self.task.train_epoch(
  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 110, in train_epoch
    return self._train_inner_loop(
  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 215, in _train_inner_loop
    loss = self.train_step(model=model, samples=samples)
  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 64, in train_step
    loss = model(samples)["loss"]
  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 886, in forward
    output = self.module(*inputs[0], **kwargs[0])
  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/models/blip2_models/blip2_qformer.py", line 171, in forward
    neg_idx = torch.multinomial(weights_t2i[b], 1).item()
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
```

**Comment by dxli94** [comment]
Hi @evelinehong , thanks for the issue.

Now fixed in https://github.com/salesforce/LAVIS/commit/0e9a3ecfa5dc668761ce6c0e3cf827578e8593c4.

Can you pull and try? Let us know how it goes.

**Comment by pixas** [comment]
> Hey I encounter this issue when I try to pre-train BLIP2 with COCO:
> 
> ```
> Traceback (most recent call last):
>  File "train.py", line 103, in <module>
>    main()
>  File "train.py", line 99, in main
>    runner.train()
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/runners/runner_base.py", line 352, in train
>    train_stats = self.train_epoch(cur_epoch)
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/runners/runner_base.py", line 411, in train_epoch
>    return self.task.train_epoch(
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 110, in train_epoch
>    return self._train_inner_loop(
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 215, in _train_inner_loop
>    loss = self.train_step(model=model, samples=samples)
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 64, in train_step
>    loss = model(samples)["loss"]
>  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
>    return forward_call(*input, **kwargs)
>  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 886, in forward
>    output = self.module(*inputs[0], **kwargs[0])
>  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
>    return forward_call(*input, **kwargs)
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/models/blip2_models/blip2_qformer.py", line 171, in forward
>    neg_idx = torch.multinomial(weights_t2i[b], 1).item()
> RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
> ```

Hello, when I run pretrain blip2 on ScienceQA, I meet the same problem. Have you solved the problem? If so, how do you solve?

**Comment by ZhaoyangLi-nju** [comment]
> Hi @evelinehong , thanks for the issue.
> 
> Now fixed in [0e9a3ec](https://github.com/salesforce/LAVIS/commit/0e9a3ecfa5dc668761ce6c0e3cf827578e8593c4).
> 
> Can you pull and try? Let us know how it goes.


the change seems to have no help in this issue?

**Comment by Victorwz** [comment]
> Hey I encounter this issue when I try to pre-train BLIP2 with COCO:
> 
> ```
> Traceback (most recent call last):
>  File "train.py", line 103, in <module>
>    main()
>  File "train.py", line 99, in main
>    runner.train()
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/runners/runner_base.py", line 352, in train
>    train_stats = self.train_epoch(cur_epoch)
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/runners/runner_base.py", line 411, in train_epoch
>    return self.task.train_epoch(
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 110, in train_epoch
>    return self._train_inner_loop(
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 215, in _train_inner_loop
>    loss = self.train_step(model=model, samples=samples)
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/tasks/base_task.py", line 64, in train_step
>    loss = model(samples)["loss"]
>  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
>    return forward_call(*input, **kwargs)
>  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/parallel/distributed.py", line 886, in forward
>    output = self.module(*inputs[0], **kwargs[0])
>  File "/gpfs/u/home/LMCG/LMCGnngn/scratch/miniconda3x86/envs/lavis/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
>    return forward_call(*input, **kwargs)
>  File "/gpfs/u/scratch/LMCG/LMCGnngn/LAVIS/lavis/models/blip2_models/blip2_qformer.py", line 171, in forward
>    neg_idx = torch.multinomial(weights_t2i[b], 1).item()
> RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
> ```

Hi Yining, I tried to reproduce BLIP-2 pre-training stage 1 with a different image-text dataset and I am faced with the same RuntimeError as yours. Have you successfully reproduce the pre-training with coco from scratch?
